pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -724,15 +724,13 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 	disks := findDisksPartitions()
 	ReportDeviceInfo.Storage = *proto.Uint64(0)
 	for _, disk := range disks {
-		size, isPart := diskmetrics.PartitionSize(disk)
+		size, _ := diskmetrics.PartitionSize(disk)
 		log.Debugf("Disk/partition %s size %d", disk, size)
 		size = RoundToMbytes(size)
 		is := info.ZInfoStorage{Device: disk, Total: size}
 		ReportDeviceInfo.StorageList = append(ReportDeviceInfo.StorageList,
 			&is)
-		if isPart {
-			ReportDeviceInfo.Storage += *proto.Uint64(size)
-		}
+		// XXX should we report whether it is a disk or a partition?
```

Comment:
That would be nice to have it in Json - We can currently tell it manually ( /dev/sda Vs /dev/sda1 )